scripts/link-checker.js

```diff
@@ -149,41 +149,79 @@ async function checkLinks() {
-                // Spoof a normal looking User-Agent to keep the servers happy
-                // See https://github.com/JustinBeckwith/linkinator/blob/main/src/index.ts
-                //
-                // To better future-proof against the ua string being
-                // responsible for any breakage, pull common, up-to-date strings
-                // from a reliable source.
-                // https://github.com/w3c/aria-practices/issues/3270
-                'User-Agent':
-                  'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Chrome/134.0.0.0 Safari/605.1.1',
-              },
-            });
-            const text = await response.text();
-            const html = HTMLParser.parse(text);
-            const ids = html
-              .querySelectorAll('[id]')
-              .map((idElement) => idElement.getAttribute('id'));
-
-            // Handle GitHub README links.
-            // These links are stored within a react-partial element
-            const reactPartial = getReactPartial(hrefOrSrc, html);
-            return {
-              ok: response.ok,
-              status: response.status,
-              ids,
-              reactPartial,
-            };
-          } catch (error) {
-            return {
-              errorMessage:
-                `Found broken external link on ${htmlPath}:${lineNumber}:${columnNumber}\n` +
-                `  ${error.stack}`,
-            };
+          const domain = new URL(externalPageLink).hostname;
+          let retryCount = 0;
+          const maxRetries = 3;
+          const baseDelay = 15;
+
+          while (retryCount < maxRetries) {
+            try {
+              const response = await nFetch(externalPageLink, {
+                headers: {
+                  // Spoof a normal looking User-Agent to keep the servers happy
+                  // See https://github.com/JustinBeckwith/linkinator/blob/main/src/index.ts
+                  //
+                  // To better future-proof against the ua string being
+                  // responsible for any breakage, pull common, up-to-date strings
+                  // from a reliable source.
+                  // https://github.com/w3c/aria-practices/issues/3270
+                  'User-Agent':
+                    'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Chrome/134.0.0.0 Safari/605.1.1',
+                },
+              });
+
+              // Handle rate limit-related statuses
+              if (
+                response.status === 403 ||
+                response.status === 429 ||
+                response.status === 503 ||
+                response.status === 508
+              ) {
+                // Found the retry-after unit returned from response headers too
+                // variable to use here, but ~15 seconds seems like a safe
+                // default
+                const retryAfter = baseDelay * 1000;
```

Comment:
Does the formula for the delay in this case need to be different than the formula for the delay in case of an exception? If not, then we could reduce duplication by simply throwing here.

Reply:
Good point, de-duped and also removed left-over retry logic in https://github.com/w3c/aria-practices/pull/3273/commits/e08b32cb4870d6a9dddc757ba46ddaeb1b3c27f3